dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/deduplication/DeduplicationController.java

```diff
@@ -189,23 +188,37 @@ public void mergePotentialDuplicate(
         TrackedEntityInstance original = getTei( potentialDuplicate.getTeiA() );
         TrackedEntityInstance duplicate = getTei( potentialDuplicate.getTeiB() );
 
+        /*
+         * We always treat the oldest tei as the original.
+         */
         if ( original.getCreated().after( duplicate.getCreated() ) )
         {
             TrackedEntityInstance t = original;
             original = duplicate;
             duplicate = t;
         }
 
-        DeduplicationMergeRequest deduplicationRequest = DeduplicationMergeRequest.builder().potentialDuplicateUid( id )
-            .mergeObject( mergeObject ).original( original ).duplicate( duplicate ).build();
+        DeduplicationMergeParams params = DeduplicationMergeParams.builder()
+            .potentialDuplicate( potentialDuplicate )
+            .mergeObject( mergeObject )
+            .original( original )
+            .duplicate( duplicate )
+            .build();
 
-        if ( MergeStrategy.MANUAL.equals( mergeStrategy ) )
+        try
         {
-            deduplicationService.manualMerge( deduplicationRequest );
+            if ( MergeStrategy.MANUAL.equals( mergeStrategy ) )
+            {
+                deduplicationService.manualMerge( params );
+            }
+            else
+            {
+                deduplicationService.autoMerge( params );
+            }
         }
-        else
+        catch ( PotentialDuplicateConflictException ex )
         {
-            deduplicationService.autoMerge( deduplicationRequest );
```

Comment:
we don't need to try catch, PotentialDuplicateConflictException has a @ResponseStatus( HttpStatus.CONFLICT )
so it is automatically caught by the controller advice, there are tests in DeduplicationControllerMvcTest for that

Reply:
Did not see this, this is a much better solution :) I'll rever the try-catch :) Thanks for the info!